frontend/src/components/Histogram.jsx

```diff
@@ -24,37 +25,83 @@ export default class Histogram extends React.Component {
   }
 
   async loadCasesData() {
+    console.log("Fetching case data");
+
+    // if there is not an available start or end date in the data use this default ones
+    const initial_date =
+      this.props.initial_date != null ? this.props.initial_date : "2020-02-20";
+    const maximum_date =
+      this.props.maximum_date != null ? this.props.maximum_date : "2020-06-23";
+
+    // converting DateFields that come from the DatePicker to string. If there is no
+    // counterfactual date use the default historical one
+    const counterfactual_first_restrictions_date =
+      this.props.counterfactual_first_restrictions_date != null
+        ? convert(this.props.counterfactual_first_restrictions_date)
+        : this.props.first_restrictions_date;
+    const counterfactual_lockdown_date =
+      this.props.counterfactual_lockdown_date != null
+        ? convert(this.props.counterfactual_lockdown_date)
+        : this.props.lockdown_date;
+
     // Retrieve real and counterfactual data in parallel
     const task = new LoadDailyCasesTask();
-    let [casesReal, casesCounterfactual] = await Promise.all([
-      task.getRealCovidCases(this.props.isoCode),
-      task.getCounterfactualCovidCases(this.props.isoCode),
+    let [casesCounterfactual, casesReal] = await Promise.all([
+      task.getCounterfactualCovidCases(
+        this.props.isoCode,
+        initial_date,
+        maximum_date,
+        counterfactual_first_restrictions_date,
+        counterfactual_lockdown_date
+      ),
+      task.getRealCovidCases(this.props.isoCode, initial_date, maximum_date),
     ]);
+
     // Combine the two datasets into a single data array
     let casesData = [];
-    for (let i = 0; i < casesReal.length; i++) {
-      const counterfactual = casesCounterfactual.find(
-        (counterfactual) => counterfactual.date === casesReal[i].date
-      );
-      let record = {
-        date: casesReal[i].date,
-        weekly_avg_real: casesReal[i].summed_avg_cases_per_million,
-        weekly_avg_counterfactual: counterfactual.summed_avg_cases_per_million,
-      };
-      casesData.push(record);
+
+    if (casesCounterfactual.length != 0) {
+      for (let i = 0; i < casesReal.length; i++) {
+        const counterfactual = casesCounterfactual.find(
+          (counterfactual) => counterfactual.date === casesReal[i].date
+        );
+        let record = {
+          date: casesReal[i].date,
+          weekly_avg_real: casesReal[i].summed_avg_cases_per_million,
+          weekly_avg_counterfactual:
+            counterfactual.summed_avg_cases_per_million,
+        };
+        casesData.push(record);
+      }
+      this.setState({ casesData: casesData });
```

Comment:
Is there a way to add the two date arrays (`counterfactual.date` and `casesReal.date`) into one and then iterate over that to produce `record`?